app/javascript/components/ShiftTable.vue

```diff
@@ -0,0 +1,83 @@
+<!-- シフトを一覧表示するテーブル -->
+
+<template>
+  <div>
+    <div v-if="filteredShiftData.length">
+      
+      <table v-if="this.admin===true">
+        <tbody>
+          <tr>
+            <th>氏名</th>
+            <th>年齢</th>
+            <th>勤務ステータス</th>
+            <th>希望出勤時間</th>
+            <th>希望退勤時間</th>
+          </tr>
+          <tr v-for="(shift) in filteredShiftData" :key="shift.id" >
+            <td><userName :key="shift.id" :userName="shift.user.name" ></userName></td>
+            <td><userAge :key="item.id" :userAge="item.user.age" ></userAge></td>
+            <td><userWorkStatus :key="shift.id" :userData="shift.user" ></userWorkStatus></td>
+            <td><requestedClockInTime :key="shift.id" :clockIn="shift.clock_in" ></requestedClockInTime></td>
+            <td><requestedClockOutTime :key="shift.id" :clockOut="shift.clock_out" ></requestedClockOutTime></td>
+          </tr>
+        </tbody>
+      </table>
+
+      <table v-else>
+        <tbody>
+          <tr>
+            <th>氏名</th>
+            <th>勤務ステータス</th>
+            <th>希望出勤時間</th>
+            <th>希望退勤時間</th>
+          </tr>
+          <tr v-for="(shift) in filteredShiftData" :key="shift.id" >
+            <td><userName :key="shift.id" :userName="shift.user.name" ></userName></td>
+            <td><userWorkStatus :key="shift.id" :userData="shift.user" ></userWorkStatus></td>
+            <td><requestedClockInTime :key="shift.id" :clockIn="shift.clock_in" ></requestedClockInTime></td>
+            <td><requestedClockOutTime :key="shift.id" :clockOut="shift.clock_out" ></requestedClockOutTime></td>
+          </tr>
+        </tbody>
+      </table>
```

Comment:
7~41行目のテーブルを出し分けている部分ですが、
年齢以外全く同じコードが二つ並んでいる点に冗長性を感じました。
例えばこのコードのままテーブルに何らかの修正を加えた時に「管理者用テーブルはちゃんと修正されているが、一般ユーザー側のテーブルは修正するのを忘れてしまった。」みたいな状況が起こる可能性があります。

例えば以下のような書き方もできるので参考にして見てください！

```suggestion
  
<table>
  <tbody>
    <tr>
      <th>氏名</th>
      <th v-if="admin">年齢</th>
      <th>勤務ステータス</th>
      <th>希望出勤時間</th>
      <th>希望退勤時間</th>
    </tr>
    <tr v-for="(shift) in filteredShiftData" :key="shift.id" >
      <td><userName :key="shift.id" :userName="shift.user.name" ></userName></td>
      <td v-if="admin"><userAge :key="shift.id" :userAge="shift.user.age" ></userAge></td>
      <td><userWorkStatus :key="shift.id" :userData="shift.user" ></userWorkStatus></td>
      <td><requestedClockInTime :key="shift.id" :clockIn="shift.clock_in" ></requestedClockInTime></td>
      <td><requestedClockOutTime :key="shift.id" :clockOut="shift.clock_out" ></requestedClockOutTime></td>
    </tr>
  </tbody>
</table>
      
```